airflow/dag_processing/processor.py

```diff
@@ -833,16 +834,20 @@ def process_file(
         :param pickle_dags: whether serialize the DAGs found in the file and
             save them to the db
         :param session: Sqlalchemy ORM Session
-        :return: number of dags found, count of import errors
+        :return: number of dags found, count of import errors, last number of db queries
         """
         self.log.info("Processing file %s for tasks to queue", file_path)
 
+        @event.listens_for(session, "do_orm_execute")
+        def _count_db_queries(orm_execute_state):
```

Comment:
I saw in tests used same approach but using "after_cursor_execute" event instead.
https://github.com/apache/airflow/blob/d5fb711ae09b06bef882a7b3b2cd20607a3671e3/tests/test_utils/asserts.py#L100

Should be the same, or there is any difference?
It would be good to know what is the difference and why there we do it differently.

Reply:
The `after_cursor_execute` event is used only with `Engine` objects. This option doesn't suit us, because we want to count queries per each file processing and for doing this we need to have unique object which should be associated with the processor's object. In my opinion Session' s object suits here well, because we create a new Session object each time when we start processing a file and then we use this object during processing for making a db's queries.
About the `Engine` object it doesn't suit, because we create an engine one time and then use it for all queries to DB in Airflow. It will be hard to distinguish what query from what part of Airflow comes.